src/mom5/ocean_bgc/ocean_generic_tracer.F90

```diff
@@ -336,9 +337,15 @@ subroutine ocean_generic_sbc(Ice_ocean_boundary_fluxes,Disd,Djsd, T_prog, runoff
     character(len=fm_string_len)      :: g_tracer_name
     character(len=fm_string_len), parameter :: sub_name = 'update_generic_tracer_sbc'
 
-    !Extract the tracer surface fields from coupler 
-    call generic_tracer_coupler_get(Ice_ocean_boundary_fluxes)
+    !Extract the tracer surface fields from coupler
+    ! dts: change to use generic_tracer_coupler_accumulate with a weight of 1. for consistency
+    ! with MOM6. Note this means that the generic_?_update_from_coupler method is no longer
+    ! called here. Instead this will be called later in ocean_generic_sbc_adjust when flux
+    ! adjustments are available
+    ! call generic_tracer_coupler_get(Ice_ocean_boundary_fluxes)
+    call generic_tracer_coupler_accumulate(Ice_ocean_boundary_fluxes, 1.)
 
+    ! dts: Is this still needed here, since this is now also done in ocean_generic_sbc_adjust?
```

Comment:
The call to `ocean_generic_sbc_adjust` is bracketed by `#ifdef USE_OCEAN_BGC`

https://github.com/ACCESS-NRI/MOM5/blob/dougiesquire/issue388-accessom-gtracer/src/mom5/ocean_tracers/ocean_tpm.F90#L1383-L1387

but the update below isn't because the call into this routine doesn't have the same pre-processor flags.

https://github.com/ACCESS-NRI/MOM5/blob/ca54c1b9d2085b1338d6aa533a03986bc1535a81/src/mom5/ocean_tracers/ocean_tpm.F90#L1306

So I'm thinking you would want to at the very least bracket the update below with `#ifndef USE_OCEAN_BGC` if you are serious about not running the section below. I haven't looked into the timing of the respective calls, so no idea if this is even feasible.

If the code block is repeated though, why not just call `ocean_generic_sbc_adjust` below?

Reply:
Sorry, I don't think I understand your comment. The call to `ocean_generic_sbc` _is_ bracketed by `#ifdef USE_OCEAN_BGC`: https://github.com/ACCESS-NRI/MOM5/blob/ca54c1b9d2085b1338d6aa533a03986bc1535a81/src/mom5/ocean_tracers/ocean_tpm.F90#L1267

Reply:
Sorry, I didn't look far enough up see the pre-processor directives. 

Basically I was trying to be helpful and do more than just point out a few 10 year old typos, but even that wasn't helpful, so I should have stuck to the typos! :)